docs/source/hive.rst

```diff
@@ -0,0 +1,94 @@
+Hive Metastore
+==============
+
+The Hive metastore relates SQL metadata to files on the Hadoop File System
+(HDFS_).  It is similar to a SQL database in that it contains information about
+SQL tables but dissimilar in that data isn't stored in Hive but remains
+ordinary files on HDFS.
+
+Odo interacts with Hive mostly through ``sqlalchemy`` and also with a bit
+of custom code due to its peculiarities.
+
+URIs
+----
+
+Hive uris match exactly SQLAlchemy connection strings with the ``hive://``
+protocol.  Additionally, Impala, another SQL database on HDFS can also connect
+to the same tables.
+::
+
+    hive://hostname
+    hive://user@hostname:port/database-name
+    hive://user@hostname:port/database-name::table-name
+
+    impala://hostname::table-name
+
+Additionally you should probably inspect docs on HDFS_ due to the tight
+integration between the two.
+
+Options
+-------
+
+Hive tables have a few non-standard options on top of normal SQL::
+
+    stored_as - File format on disk like TEXTFILE, PARQUET, ORC
+    path - Absolute path to file location on HDFS
+    external=True - Whether to keep the file external or move it to Hive
+        directory
+
+See `Hive DDL`_ docs for more information.
+
+
+Conversions
+-----------
+
+We commonly load CSV files in to Hive, either from HDFS or from local disk on
+one of the machines that comprise the HDFS cluster::
+
+    HDFS(Directory(CSV)) -> Hive
+    SSH(Directory(CSV)) -> Hive
+    SSH(CSV) -> Hive
+
+Additionally we can use Hive to efficiently migrate this data to new data in a
+different format::
+
+    Hive -> Hive
+
+And as with all SQL systems through SQLAlchemy we can convert a Hive table to a
+Python Iterator, though this is somewhat slow::
+
+    Hive -> Iterator
+
+
+Impala
```

Comment:
I wonder if a small list of technologies that odo handles that work with hive might be nice.